### DISCUSSION TITLE
700 Series Dimmer issue when Switch Mode is Enabled

### DISCUSSION THREAD

**Discussion by DavinKD** [zwave-js/zwave-js#7167] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
When a newer 700 series dimmer is set to Switch Mode, when you turn on via Home Assistant or Zwave-Js UI, the light cannot be turned off via HA or Zwave-Js-Ui.  The light also does not turn off when the bottom paddle is touched as well.  However, if you touch the top paddle then the bottom paddle, the light will turn off.  When switch mode is disabled, all works fine.

I don't recall having this issue when the switch was on SmartThings.

If you turn the light on mechanically with the top paddle, it turns off normally using the bottom paddle.

What did you expect to happen instead?
When HA or Zwave-JS UI commands the light to turn off, it should turn off.

  1. Ensure that Switch Mode enabled, (Parameter 16) is set to 1
  2. When the switch is off, go to Home Assistant and toggle the switch on.  The light comes on.
  3. Then toggle switch off, light stays on
  4. Touch bottom paddle on switch, light stays on, but indicator flashes briefly. (this can be repeated several times)
  5. Touch top paddle - Nothing appears to happen
  6. Touch bottom paddle - Light turns off.

Anything else we should know?
This only seems to be an issue on a new switch where "Switch Mode" appears in the UI.  For older ones with it enabled using Parameter 16, I have no issues.

The switch in question is an Ultrapro 59350 / 59372 / 59373 / ZWA3016.  The firmware is version 1.23.2.

### Software versions

Driver (node-zwave-js): 13.2.0

Z-Wave JS UI: 9.18.1

Home Assistant Z-Wave Integration: Sorry, not sure how to tell.  HA is version 24.9.1

Home Assistant Z-Wave JS Addon: N/A running separately.

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...

### Device information

Manufacturer: Ultrapro
Model name: 59350 / 59372 / 59373 / ZWA3016.
Node ID: 30

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[logs.zip](https://github.com/user-attachments/files/17004437/logs.zip)

[a comment by an automated account is not shown]

**Comment by DavinKD** [comment]
I have a second dimmer that wasn't in the wall yet that I loaded into SmartThings, and it seems to work fine there.

**Comment by AlCalzone** [comment]
I'm not entirely sure what switch mode does, and the available manuals online aren't particularly helpful.

It is possible that that configuration changes the device capabilities from a Multilevel to a Binary Switch, which means you may have to exclude and include the device again for the changes to take effect.

**Reply by DavinKD** [reply]
Sorry for the slow response.  Switch Mode causes the switch to turn on to full brightness and off instantly when controlled manually, but also allows for dimming via the Zwave commands.  As I also said above, older dimmers (I have as old as 2016 versions) also support this through manually setting Parameter 16 to a value of 1.  Those work fine.  It's just the new one that has a dedicated setting for "Switch Mode" that has the issue.

It does remain as a Multilevel switch.

**Reply by AlCalzone** [reply]
> It does remain as a Multilevel switch.

Even after excluding and re-including?

**Reply by DavinKD** [reply]
Correct.  I just tested it again by excluding and re-including.  It does remain as a Multilevel switch and still exhibits the issue.

**Reply by AlCalzone** [reply]
Then I suppose it's a question for the manufacturer support.

**Reply by DavinKD** [reply]
Ok, but I did want to reiterate, it works fine in SmartThings which is why I believe it's a software issue on the Zwave-JS side and not a firmware issue with the switch.  For now, I've disabled Switch Mode on this particular switch and I only have one in production, so not a huge deal.

**Reply by AlCalzone** [reply]
From the Z-Wave JS perspective, the "Switch Mode" is a number like any other configuration parameter and that's it. It does not know or care what configuration parameters do.

If you tell it to turn a dimmer on, it sends a `Multilevel Switch Set` with target value 99 or 255 (0xff). If you tell it to turn it off, it sends the same with target value 0:

![grafik](https://github.com/user-attachments/assets/e18c6f7b-7ffd-4053-ac84-63996781a080)

If a device does not turn off in response to being set to 0, then it does not behave as specified, which is a device bug. Especially when local control doesn't work correctly either if I understood you correctly?
Your log also shows a few instances where the device reports that it is transitioning from 99 to 0 with a remaining time of 0s, but it doesn't seem to actually reach 0.

**Reply by AlCalzone** [reply]
FWIW, there seems to be a firmware update available on the Jasco repository:
https://github.com/jascoproducts/firmware/tree/main/zwave/UltraPro/ZWA3016/59350%20-%20In-Wall%20Paddle%20Dimmer%2C%20700S

Might be worth a try, although I don't see anything like this issue mentioned.

**Comment by AlCalzone** [comment]
Ok, I have a hunch what could be happening here. Z-Wave JS uses Supervision where possible, AFAIK SmartThings does not. Some devices have issues in their Supervision implementation and behave erratically. Maybe this is the case here too, but only in Switch Mode for some reason.

If you want to try, please follow the instructions at https://zwave-js.github.io/zwave-js-ui/#/device-support/custom-device-files using
[this file](https://github.com/user-attachments/files/17094398/59350_59372_59373_zwa3016.json).

After restarting Z-Wave JS UI, and re-interviewing the device, try controlling it in switch mode again. Let me know if anything changes.

**Reply by DavinKD** [reply]
I'll try this out tonight.

**Reply by DavinKD** [reply]
Unfortunately, it didn't help.

**Reply by DavinKD** [reply]
Does the custom json work on Windows using the pre-compiled .exe?  Reason I ask is that I modified the JSON myself and changed it a little and it doesn't seem to be picking up the changes even when I stop/start the .exe.

**Reply by AlCalzone** [reply]
Oh, I'm not sure about that tbh. The driver log should indicate at startup that a "user-defined" config was loaded.
@robertsLando do you know?

**Reply by robertsLando** [reply]
> Does the custom json work on Windows using the pre-compiled .exe? Reason I ask is that I modified the JSON myself and changed it a little and it doesn't seem to be picking up the changes even when I stop/start the .exe.

It does, you just have to respect the file location, you should place it under the store folder (did you changed it?) in a folder named `config` like explained in docs, then on driver logs you should see that so please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Reply by DavinKD** [reply]
[zwavejs_2024-09-25.log](https://github.com/user-attachments/files/17133747/zwavejs_2024-09-25.log)

I do see this in the log

2024-09-25T15:04:22.357Z CNTRLR   [Node 047] Embedded device config loaded

This is the device in question.

![image](https://github.com/user-attachments/assets/ea3f4ed3-0785-41cc-9bc4-21358b12e1e4)

**Reply by robertsLando** [reply]
What path do you have here in settings?

![2024-09-25_17-26](https://github.com/user-attachments/assets/2cf5e35e-f8bb-40e0-95b2-7c9f4e9a8d4f)

**Reply by DavinKD** [reply]
Aha, that may be the issue.

C:\Users\davin\AppData\Roaming\npm\node_modules\zwave-js-ui\store/config

**Reply by robertsLando** [reply]
@DavinKD fix that and save, then check logs, you should see `User-provided device config loaded` now

**Reply by DavinKD** [reply]
Ok, I can confirm it's using the custom driver now, but the original issue is still there. I'm with you now that it's probably a firmware issue.  Just not sure how ST was getting around it.

**Reply by DavinKD** [reply]
@robertsLando , ps.  Love the product.
